Src/Support/Google.Apis.Tests/Apis/Requests/BatchRequestTest.cs

```diff
@@ -446,6 +447,108 @@ public void BatchUrl()
+}");
+
+            using (var service = new MockClientService(
+                new BaseClientService.Initializer()
+                {
+                    HttpClientFactory = new MockHttpClientFactory(new BatchEndpointErrorMessageHandler(jsonErrorContent))
+                },
+                "http://sample.com"))
+            {
+                var batchRequest = new BatchRequest(service);
+                var request = new TestClientServiceRequest(service, new MockRequest
+                {
+                    ETag = "\"100\"",
+                    Name = "Name1"
+                });
+                
+                batchRequest.Queue<MockResponse>( request, (c, e, i, m) =>
+                    Assert.True(false, "The batch endpoint call should have failed. Callbacks for individual requests shouldn't be called."));
+
+                HttpRequestException outerException = await Assert.ThrowsAsync<HttpRequestException>(() => batchRequest.ExecuteAsync());
+
+                Assert.Contains("400", outerException.Message);
+                Assert.Contains("Bad Request", outerException.Message);
+
+                GoogleApiException innerException = Assert.IsType<GoogleApiException>(outerException.InnerException);
+
+                Assert.Equal(HttpStatusCode.BadRequest, innerException.HttpStatusCode);
+                Assert.NotNull(innerException.Error);
+
+                RequestError requestError = innerException.Error;
+                Assert.Equal(jsonErrorContent, requestError.ErrorResponseContent);
+                Assert.Equal(400, requestError.Code);
+
+                SingleError singleError = Assert.Single(requestError.Errors);
+                Assert.Equal("global", singleError.Domain);
+                Assert.Equal("badRequest", singleError.Reason);
+            }
+        }
+
+        [Theory]
+        [InlineData("Invalid JSON")]
+        [InlineData(null)]
+        public async Task BatchEndpointNonJsonContent(string content)
+        {
+            using (var service = new MockClientService(
+                new BaseClientService.Initializer()
+                {
+                    HttpClientFactory = new MockHttpClientFactory(new BatchEndpointErrorMessageHandler(content))
+                },
+                "http://sample.com"))
+            {
+                var batchRequest = new BatchRequest(service);
+                var request = new TestClientServiceRequest(service, new MockRequest
+                {
+                    ETag = "\"100\"",
+                    Name = "Name1"
+                });
+
+                batchRequest.Queue<MockResponse>(request, (c, e, i, m) =>
+                   Assert.True(false, "The batch endpoint call should have failed. Callbacks for individual requests shouldn't be called."));
+
+                HttpRequestException outerException = await Assert.ThrowsAsync<HttpRequestException>(() => batchRequest.ExecuteAsync());
+
+                Assert.Contains("400", outerException.Message);
```

Comment:
Should the content be present anywhere in the exception?

Reply:
(Doesn't look like we're adding it explicitly. I think we include it for non-batch requests, don't we?)

Reply:
Yes, for non batch, we throw a GoogleApiException for these cases that contain the content if there was some. We can include that GoogleApiException as the inner exception, let me try that.